src/app/state/app.state.ts

```diff
@@ -22,84 +21,195 @@ export class AppState {
         'Leaphy Original Extension', 'Leaphy Extra Extension', 'Servo', 'Adafruit GFX Library', 'Adafruit SSD1306', 'Adafruit LSM9DS1 Library',
         'Adafruit Unified Sensor', 'List', 'Adafruit SGP30 Sensor', 'Adafruit_VL53L0X', 'Adafruit BMP280 Library', 'TM1637', 'LedControl'
     ]
-    private static leaphyOriginalRobotType = new RobotType('l_original_uno', Stk500v1, 'Leaphy Original', 'orig.svg', 'Arduino UNO', 'arduino:avr:uno', 'hex', 'arduino:avr',
+    private static leaphyOriginalRobotType = new RobotType(
+        'l_original_uno',
+        Avrdude,
+        'Leaphy Original',
+        'orig.svg',
+        'atmega328p',
+        'arduino:avr:uno',
+        'hex',
+        'arduino:avr',
         AppState.defaultLibraries.concat(['QMC5883LCompass', 'Arduino_APDS9960']), true, {
             showLeaphyActuators: true,
             showLeaphyOperators: false,
             showLeaphySensors: true,
         },
     );
-    private static leaphyOriginalNanoRobotType = new RobotType('l_original_nano', Stk500v1, 'Original Nano', 'orig.svg', 'Arduino Nano', 'arduino:avr:nano', 'hex', 'arduino:avr',
+    private static leaphyOriginalNanoRobotType = new RobotType(
+        'l_original_nano',
+        Avrdude,
+        'Original Nano',
+        'orig.svg',
+        'atmega328p',
+        'arduino:avr:nano',
+        'hex',
+        'arduino:avr',
         AppState.defaultLibraries.concat(['QMC5883LCompass', 'Arduino_APDS9960']), true, {
             showLeaphyActuators: true,
             showLeaphyOperators: true,
             showLeaphySensors: true,
         },
     );
-    private static leaphyOriginalNanoESP32RobotType = new RobotType('l_original_nano_esp32', DFU, 'Original Nano ESP32', 'orig.svg', 'Arduino Nano ESP32', 'arduino:esp32:nano_nora', 'bin', 'arduino:esp32',
+    private static leaphyOriginalNanoESP32RobotType = new RobotType(
+        'l_original_nano_esp32',
+        DFU,
+        'Original Nano ESP32',
+        'orig.svg',
+        'NOT_SET', // for other uploaders then avrdude NOT_SET is used
```

Comment:
Convert the protocol parameter to an object with optional metadata to clean up arguments which aren't necessary for other uploaders